code/modules/mob/living/carbon/xenomorph/evolution.dm

```diff
@@ -311,7 +241,7 @@
 			qdel(new_xeno)
 		return
 	new_xeno.upgrade_stored = upgrade_stored
-	while(new_xeno.upgrade_stored >= new_xeno.xeno_caste.upgrade_threshold && new_xeno.upgrade_possible())
+	while(new_xeno.upgrade_stored >= new_xeno?.xeno_caste.upgrade_threshold && new_xeno.upgrade_possible())
```

Comment:
It already makes sure that new xeno exists seven lines up, why the uncertainty here? Upgrading shouldn't null it.

Reply:
Had runtimes in here at some point, the uncertainty is on them having a xeno caste, but I wasn't too sure how the ?. operator worked, whether it checks the former or the latter, I assume now it checks the latter